### ISSUE TITLE
TR-DPO : Why is the loss not changing at all, and reward/accuracies and reward/margins always = 0?

### ISSUE THREAD

**Issue by GritLs** [huggingface/trl#1843]
### Problem discription
I am encountering an issue while using TR-DPO for training. The log output shows that the loss does not change at all, and both reward/accuracies and reward/margins are always equal to 0. Could you please help me understand why this is happening?
### Minimal reproduce
```
from transformers import (
    AutoModelForCausalLM,
    AutoTokenizer,
)
from trl import DPOTrainer,DPOConfig
from datasets import Dataset
from accelerate import Accelerator
dpo_dataset_dict = {
    "prompt": [
        "hello",
        "how are you",
        "What is your name?",
        "What is your name?",
        "Which is the best programming language?",
        "Which is the best programming language?",
        "Which is the best programming language?",
    ],
    "chosen": [
        "hi nice to meet you",
        "I am fine",
        "My name is Mary",
        "My name is Mary",
        "Python",
        "Python",
        "Java",
    ],
    "rejected": [
        "leave me alone",
        "I am not fine",
        "Whats it to you?",
        "I dont have a name",
        "Javascript",
        "C++",
        "C++",
    ],
}

for key in dpo_dataset_dict.keys():
    for i in range(12):
        dpo_dataset_dict[key].extend(dpo_dataset_dict[key])

model = AutoModelForCausalLM.from_pretrained("retriever_model/gpt2")
tokenizer = AutoTokenizer.from_pretrained("retriever_model/gpt2")
tokenizer.pad_token = tokenizer.eos_token
train_dataset = Dataset.from_dict(dpo_dataset_dict)
training_args = DPOConfig(
    beta=0.1,
    sync_ref_model=True,
    output_dir="debug/rubbish/",
    logging_steps=5
)
dpo_trainer = DPOTrainer(
    model,
    ref_model=model,
    args=training_args,
    train_dataset=train_dataset,
    tokenizer=tokenizer,  # for visual language models, use tokenizer=processor instead
)

dpo_trainer.train()
```
```
CUDA_VISIBLE_DEVICES=0 python path/to/script.py
```

This is just a small part of the log. Even after training for thousands of steps, the loss still hasn't changed at all.
![image](https://github.com/user-attachments/assets/d70613ef-f6a2-4c81-b3b6-598357245463)

**Comment by GritLs** [comment]
**Problem Cause:**

The `ref_model` and the `target_model` cannot reference the same loaded model. If they do, as the `target_model` gets optimized, the `ref_model` will also be optimized. Consequently, the log probability ratio of the two models will always remain unchanged, equating to zero.

**Comment by fadybaly** [comment]
hey @GritLs I'm having the same issue you did with the same exact loss values. although in the documentation it's mentioned that setting `ref_model = None` automatically creates a copy of the model. so I'm not sure where the issue is.
how did you end up solving this?
